client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -138,16 +155,32 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 	const connectComponent = connect(
 		state => {
 			const siteId = getSelectedSiteId( state );
-			const isRequestingSettings = isRequestingSiteSettings( state, siteId );
-			const isSavingSettings = isSavingSiteSettings( state, siteId );
-			const isSaveRequestSuccessful = isSiteSettingsSaveSuccessful( state, siteId );
-			const settings = getSiteSettings( state, siteId );
+			let isRequestingSettings = isRequestingSiteSettings( state, siteId );
+			let isSavingSettings = isSavingSiteSettings( state, siteId );
+			let isSaveRequestSuccessful = isSiteSettingsSaveSuccessful( state, siteId );
+			let settings = getSiteSettings( state, siteId );
+			const settingsFields = {
+				site: keys( settings ),
+			};
+
+			const isJetpack = isJetpackSite( state, siteId );
+			if ( isJetpack ) {
+				const jetpackSettings = getJetpackSettings( state, siteId );
+				isRequestingSettings = isRequestingSettings || isRequestingJetpackSettings( state, siteId );
+				isSavingSettings = isSavingSettings || isUpdatingJetpackSettings( state, siteId );
+				isSaveRequestSuccessful = isSaveRequestSuccessful && isJetpackSettingsSaveSuccessful( state, siteId );
+				settings = { ...settings, ...jetpackSettings };
+				settingsFields.jetpack = keys( jetpackSettings );
+			}
+
 			return {
-				isRequestingSettings: isRequestingSettings && ! settings,
+				isRequestingSettings,
```

Comment:
Makes sense. I've taken care of this, see https://github.com/Automattic/wp-calypso/pull/10697/commits/d01101fa65bd267ee59158ecac48610b98102b13.